### PR TITLE
More distros

### PR DESCRIPTION
**Pull request by dcermak**
Add tests for Archlinux, Ubuntu, Debian, CentOS 7 & 8.

Currently, there's still a lot of failing tests, so this needs further work…

### CODE REVIEW

File: launcher/constants.py [hunk 1]
```diff
@@ -98,6 +98,18 @@
 SLE_DISTRI = "sle"
 SLE_15_VERSION = "15"
 
+CENTOS_DISTRI = "centos"
+CENTOS_7_VERSION = "7"
+CENTOS_8_VERSION = "8"
```
Comment (Conan-Kudo) [inline launcher/constants.py:103]: Can we add CentOS 9 too?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: launcher/constants.py (hunk 1 above)
Comment (dcermak) [reply launcher/constants.py:103]: We can, if we had test images in staging.

Comment (Conan-Kudo) [reply launcher/constants.py:103]: I think @schaefi made those, didn't he?

Comment (dcermak) [reply launcher/constants.py:103]: It looks like we could actually create these as the repos for CentOS Stream 9 are already there: https://build.opensuse.org/project/show/CentOS:CentOS-9:Stream.
However, this is a bit out of scope of this PR :wink:

**Comment by dcermak** [comment]
I've made a full test run for Ubuntu, Debian, CentOS and Archlinux and there are now only very few failures left, exclusively on the Archlinux side:
1. see https://github.com/OSInside/kiwi/issues/2005
2. the KIS images for Archlinux cannot be booted by openQA (the capability to do this has simply never been implemented). I would therefore vote to simply omit these until this functionality is added to openQA (if it ever will…)

**Review by Conan-Kudo — APPROVED** [review]

**Review by schaefi — APPROVED** [review]
very nice extending the matrix change :+1:

File: launcher/constants.py (hunk 1 above)
Comment (schaefi) [reply launcher/constants.py:103]: The stream9 tests are currently not building, we need to fix the systemdeps requires and there is also a package missing  in stream9 for obs (pv)

**Comment by schaefi** [comment]
@dcermak This is great work, thanks. I think you should merge it and we will come up with a new PR once stream9 tests builds

**Comment by dcermak** [comment]
> @dcermak This is great work, thanks. I think you should merge it and we will come up with a new PR once stream9 tests builds

Well, the archlinux tests are still mostly broken atm, so that needs to get fixed before we can proceed here.

File: launcher/constants.py (hunk 1 above)
Comment (schaefi) [reply launcher/constants.py:103]: stream 9 tests are done

**Comment by schaefi** [comment]
> > @dcermak This is great work, thanks. I think you should merge it and we will come up with a new PR once stream9 tests builds
> 
> Well, the archlinux tests are still mostly broken atm, so that needs to get fixed before we can proceed here.

ok, I think @davidcassany wanted to look at the Arch issues which are related to dracut iirc

**Comment by Conan-Kudo** [comment]
If Arch Linux is still failing, do we want to disable it until @davidcassany can get things fixed?

**Comment by schaefi** [comment]
@dcermak with regards to @Conan-Kudo suggestion to disable the Arch tests and your followup commits, can this one be merged now ?

Thanks

**Comment by schaefi** [comment]
@dcermak This is sitting for a long time. Can you please give an update on the path forward ? Thanks

**Pull request merged**